File: src/main/java/com/demo/medapp/dtos/DoctorDto.java
package com.demo.medapp.dtos;

import com.demo.medapp.models.Location;

public record DoctorDto(
        String firstName,
        String lastName,
        String email,
        String phoneNumber,
        String licenseNumber,
        String speciality,
        LocationDto location,
        boolean isValidated

) {
}

File: src/app/models/doctor-dto.ts
import { LocationDto } from './location-dto.model';

export interface Doctor {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  speciality: string;
  location: LocationDto;
  licenseNumber: string;
  isValidated: boolean;
  isPending: boolean;
}